lms/djangoapps/bookmarks/tests/test_views.py

```diff
@@ -0,0 +1,462 @@
+        """
+        self.assertEqual(response_data['id'], "%s,%s" % (self.user.username, unicode(bookmark.usage_key)))
+        self.assertEqual(response_data['usage_id'], unicode(bookmark.usage_key))
+        self.assertEqual(response_data['course_id'], unicode(bookmark.course_key))
+        self.assertIsNotNone(response_data["created"])
+
+        if optional_fields:
+            self.assertEqual(response_data['path'], bookmark.path)
+            self.assertEqual(response_data['display_name'], bookmark.display_name)
+
+    def send_get(self, client, url, query_parameters=None, expected_status=200):
+        """
+        Helper method for sending a GET to the server. Verifies the expected status and returns the response.
+        """
+        url = url + '?' + query_parameters if query_parameters else url
+        response = client.get(url)
+        self.assertEqual(expected_status, response.status_code)
+        return response
+
+    def send_post(self, client, url, json_data, content_type="application/json", expected_status=201):
+        """
+        Helper method for sending a POST to the server. Verifies the expected status and returns the response.
+        """
+        url = url + '?course_id={}'.format(self.course_id)
+        response = client.post(url, data=json.dumps(json_data), content_type=content_type)
+        self.assertEqual(expected_status, response.status_code)
+        return response
+
+    def send_delete(self, client, url, expected_status=204):
+        """
+        Helper method for sending a DELETE to the server. Verifies the expected status and returns the response.
+        """
+        response = client.delete(url)
+        self.assertEqual(expected_status, response.status_code)
+        return response
+
+
+class BookmarksViewTests(BookmarksViewTestsMixin):
+    """
+    This contains the tests for GET & POST methods of bookmark.views.BookmarksView class
+    GET /api/bookmarks/v0/bookmarks/?course_id={course_id1}
+    POST /api/bookmarks/v0/bookmarks/?course_id={course_id1}
+    """
+    def test_get_bookmarks_successfully(self):
+        """
+        Test that requesting bookmarks for a course returns records successfully in
+        expected order without optional fields.
+        """
+        query_parameters = 'course_id={}'.format(self.course_id)
+        response = self.send_get(client=self.client, url=reverse('bookmarks'), query_parameters=query_parameters)
+
+        bookmarks = response.data['results']
+
+        self.assertEqual(len(bookmarks), 2)
+        self.assertEqual(response.data['count'], 2)
+        self.assertEqual(response.data['num_pages'], 1)
+
+        # As bookmarks are sorted by -created so we will compare in that order.
+        self.assert_valid_bookmark_response(bookmarks[0], self.bookmark_2)
+        self.assert_valid_bookmark_response(bookmarks[1], self.bookmark_1)
+
+    def test_get_bookmarks_with_optional_fields(self):
```

Comment:
Nit: you could use DDT to provide the value of optional_fields and collapse this test and the previous into just one test that is run twice.